eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -140,7 +140,7 @@ object Upstream {
   sealed trait Hot extends Upstream
   object Hot {
     /** Our node is forwarding a single incoming HTLC. */
-    case class Channel(add: UpdateAddHtlc, receivedAt: TimestampMilli) extends Hot {
+    case class Channel(add: UpdateAddHtlc, receivedAt: TimestampMilli, receivedFrom: PublicKey) extends Hot {
```

Comment:
I was interested in adding this as well for updates to the `AuditDb`. Tracking the incoming peer in `Hot` upstreams is valuable on its own, can you extract that to its own PR?

Reply:
https://github.com/ACINQ/eclair/pull/2883